signature/mechanism_test.go

```diff
@@ -1,3 +1,6 @@
+//go:build cgo || containers_image_openpgp
+// +build cgo containers_image_openpgp
```

Comment:
If the goal is to make tests pass with `containers_image_disable_signing` (and I do think those tests should pass, because parts of c/image/signature are completely independent of the simple signing implementation), this doesn’t do that, and from a quick check it would be a rather larger effort — because individual expected results would have to be made conditional, correctly.

Reply:
The goal was to make `go test` pass by excluding the signing tests unless a signing mechanism is compiled in.

Why would we need to run the signing tests when the entire signature mechanism is not compiled in (in the case of containers_image_disable_signing) and will always return errors?